go/libraries/doltcore/sqle/writer/prolly_index_writer.go

```diff
@@ -422,7 +438,13 @@ func FormatKeyForUniqKeyErr(key val.Tuple, d val.TupleDesc) string {
 			sb.WriteString(",")
 		}
 		seenOne = true
-		sb.WriteString(d.FormatValue(i, key.GetField(i)))
+		switch d.Types[i].Enc {
+		// address encodings should be printed as strings
+		case val.BytesAddrEnc, val.StringAddrEnc:
```

Comment:
Do we need to include `jsonAddrEnc`? 

I'm not sure if we should print out the blob content as a string... perhaps as hex bytes or something? I'd guess that more often than not, text in a `BLOB` is binary data and won't be valid string data.

Reply:
MySQL's converts blob types to string representation in just the error message.
```
mysql> insert into t values ('hello');
ERROR: 1062 (23000): Duplicate entry 'hello' for key 't.i'

mysql> insert into t values (0x68656c6c6f);
ERROR: 1062 (23000): Duplicate entry 'hello' for key 't.i'
```

JSON indexes must be over generated columns, so I don't think that applies here